src/shmem_node_util.h

```diff
@@ -0,0 +1,54 @@
+/* -*- C -*-
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S.  Government
+ * retains certain rights in this software.
+ *
+ * Copyright (c) 2018 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+#ifndef SHMEM_NODE_UTIL_H
+#define SHMEM_NODE_UTIL_H
+
+#include <limits.h>
+#include <sys/param.h>
+
+#ifdef MAXHOSTNAMELEN
+#define SHMEM_INTERNAL_MAX_HOSTNAME_LEN MAXHOSTNAMELEN
+#else
+#define SHMEM_INTERNAL_MAX_HOSTNAME_LEN HOST_NAME_MAX
+#endif
+
+extern char *shmem_internal_location_array;
```

Comment:
Given the the usage model is signed (0/-1), it would be safer for the type of this array to be `signed char`.  From the C99 spec:
> If the value of an object of type char is treated as a signed integer when used in an
expression, the value of CHAR_MIN shall be the same as that of SCHAR_MIN and the
value of CHAR_MAX shall be the same as that of SCHAR_MAX. Otherwise, the value of
CHAR_MIN shall be 0 and the value of CHAR_MAX shall be the same as that of
UCHAR_MAX.